src/app/utils/PossibleKnightMoves.ts

```diff
@@ -0,0 +1,66 @@
+import { CordWithMoveType, IChessBoard, Piece, PieceType, Side, Cord } from '../domain/basicChessTypes';
+import { ChessBoard } from '../infrastructure/ChessBoard';
+// import { ChessEngine } from '../infrastructure/ChessEngine';
+
+const chessboard = new ChessBoard();
+chessboard.board = [
+    [null, null, null, null, null, null, null, null],
+    [null, null, null, null, null, null, null, null],
+    [null, null, null, null, null, null, null, null],
+    [null, null, null, null, null, null, null, null],
+    [null, null, null, null, null, null, null, null],
+    [null, null, null, null, null, null, null, null],
+    [null, null, null, null, null, null, null, null],
+    [null, null, null, null, null, null, null, null],
+];
+
+const knight: Piece = {
+    figType: PieceType.Knight,
+    cord: { x: 1, y: 0 },
+    side: Side.Black,
+    isMoved: true,
+};
+
+chessboard.board[knight.cord.x][knight.cord.y] = knight;
+
+function getPossibleMovesForKnight(cord: Cord, boardState: IChessBoard) {
+    const possibleMoves = [];
+    const possibleCombinations = [
+        [1, 2],
+        [-1, 2],
+        [1, -2],
+        [-1, -2],
+    ];
+
+    for (const combination in possibleCombinations) {
```

Comment:
Pętla powinna być `for of` zamiast `for in` ,bo chcemy iterować po wartość ([1, 2]), a nie po kluczach. Te klucze są stringami i niepoprawne koordynaty są wrzucane do tablicy.